src/runtime/storage/pouchdb/pouch-db-collection.ts

```diff
@@ -335,7 +334,8 @@ export class PouchDbCollection extends PouchDbStorageProvider implements Collect
     // Keep retrying the operation until it succeeds.
     while (1) {
       // TODO(lindner): add backoff and error out if this goes on for too long
-      let doc: PouchDB.Core.ExistingDocument<CollectionStorage>;
+      // tslint:disable-next-line: no-any
+      let doc: any;
```

Comment:
Why has this lost the stronger typing?

Reply:
Fixed

Reply:
Wait no, there were a bunch of errors. We assign a whole bunch of things to doc that aren't on the type PouchDB.Core.ExistingDocument<CollectionStorage>. e.g. line 361:

```
        doc = {
          _id: this.pouchDbKey.location,
          referenceMode: this.referenceMode,
          type: this.type.toLiteral(),
        };
```

I'm not sure what type this actually should be. I left it as `any` in order to get ti to compile (it was effectively any beforehand anyway, since we didn't have pouch type defs). Any ideas?

Reply:
If there's a known set of fields that we work with on the pouch doc object, we can just define an interface to capture that.

Reply:
Done. The missing model and version fields get updated below, so I've set them to null temporarily to make the type checker happy. I don't know how to set _rev though (I set it to null too)